chain/near.ts

```diff
@@ -0,0 +1,416 @@
+      return changetype<AddKeyAction>(this.data as u32)
+    }
+
+    toDeleteKey(): DeleteKeyAction {
+      assert(this.kind == ActionKind.DELETE_KEY, 'ActionValue is not a \'DeleteKey\'.')
+      return changetype<DeleteKeyAction>(this.data as u32)
+    }
+
+    toDeleteAccount(): DeleteAccountAction {
+      assert(this.kind == ActionKind.DELETE_ACCOUNT, 'ActionValue is not a \'DeleteAccount\'.')
+      return changetype<DeleteAccountAction>(this.data as u32)
+    }
+
+    static fromCreateAccount(input: CreateAccountAction): ActionValue {
+      return new ActionValue(ActionKind.CREATE_ACCOUNT, changetype<u32>(input))
+    }
+
+    static fromDeployContract(input: DeployContractAction): ActionValue {
+      return new ActionValue(ActionKind.DEPLOY_CONTRACT, changetype<u32>(input))
+    }
+
+    static fromFunctionCall(input: FunctionCallAction): ActionValue {
+      return new ActionValue(ActionKind.FUNCTION_CALL, changetype<u32>(input))
+    }
+
+    static fromTransfer(input: TransferAction): ActionValue {
+      return new ActionValue(ActionKind.TRANSFER, changetype<u32>(input))
+    }
+
+    static fromStake(input: StakeAction): ActionValue {
+      return new ActionValue(ActionKind.STAKE, changetype<u32>(input))
+    }
+
+    static fromAddKey(input: AddKeyAction): ActionValue {
+      return new ActionValue(ActionKind.ADD_KEY, changetype<u32>(input))
+    }
+
+    static fromDeleteKey(input: DeleteKeyAction): ActionValue {
+      return new ActionValue(ActionKind.DELETE_KEY, changetype<u32>(input))
+    }
+
+    static fromDeleteAccount(input: DeleteAccountAction): ActionValue {
+      return new ActionValue(ActionKind.DELETE_ACCOUNT, changetype<u32>(input))
+    }
+  }
+
+  // We don't map ReceiptData
+  export class ActionReceipt {
+    constructor(
+      // Receipt fields
+      public predecessorId: string,
+      public receiverId: string,
+      public id: CryptoHash,
+
+      // ReceiptAction fields
+      public signerId: string,
+      public signerPublicKey: PublicKey,
+      public gasPrice: BigInt,
+      public outputDataReceivers: Array<DataReceiver>,
+      public inputDataIds: Array<CryptoHash>,
+      public actions: Array<ActionValue>,
+    ) {}
+  }
+
+  export enum SuccessStatusKind {
+    VALUE = 0,
+    RECEIPT_ID = 1,
+  }
+
+  // Doesn't have Value suffix because it has
+  // VALUE variant/kind, that would be confusing.
+  export class SuccessStatus {
+    constructor(
+      public kind: SuccessStatusKind,
+      public data: Payload,
+    ) {}
+
+    toValue(): string {
+      assert(this.kind == SuccessStatusKind.VALUE, 'SuccessStatus is not a \'Value\'.')
+      return changetype<string>(this.data as u32)
+    }
+
+    toReceiptId(): CryptoHash {
+      assert(this.kind == SuccessStatusKind.RECEIPT_ID, 'SuccessStatus is not a \'ReceiptId\'.')
+      return changetype<CryptoHash>(this.data as u32)
+    }
+
+    static fromValue(input: string): SuccessStatus {
```

Comment:
A `SuccessValueExecutionStatus` is always base64 encoded as a string and represents a series of Bytes. We should also turn this one into a `Bytes` that I will base64 decode in `graph-node` for now.

Reply:
Fixed!